dell/csi-baremetal-operator
# [ISSUE-572] ability to set mem/cpu for csi containers

## Purpose
### Issue dell/csi-baremetal#572

- Add `Resources *corev1.ResourceRequirements` to all components `api/v1/components`
- Add follow tree to each csi component in csi-baremetal-deployment and value.yaml:

```
      resources:
        requests:
          memory: 
          cpu: 
        limits:
          memory: 
          cpu:
```

## PR checklist

## Testing
Check current resources:
1. deploy csi-operator from master branch
```
helm install csi-baremetal-operator http://asdrepo.isus.emc.com:8081/artifactory/atlantic-helm-build/csi-operator/${CSI_OPERATOR}/csi-baremetal-operator-${CSI_OPERATOR}.tgz --set image.tag=${CSI_OPERATOR} --set global.registry=asdrepo.isus.emc.com:9042 --set log.level=debug
```

2. deploy csi driver:
```
helm install csi-baremetal http://asdrepo.isus.emc.com:8081/artifactory/atlantic-helm-build/csi-operator/${CSI_OPERATOR}/csi-baremetal-deployment-${CSI_OPERATOR}.tgz --set image.tag=${CSI_VERSION} --set global.registry=asdrepo.isus.emc.com:9042 --set log.level=debug --set feature.extender=true  --set driver.drivemgr.type=halmgr  --set driver.alerts.mountConfig=true --set platform=vanilla
```

3. check csi containers resources:
```
$ kubectl get pod csi-baremetal-controller-64d9bd7bb5-mbv5x --output=yaml | grep -A 6 resources
    resources: {}
    ...
--
    resources: {}
    ...
--
    resources: {}
	...
--
    resources: {}
    ...
$ kubectl get pod csi-baremetal-node-controller-6d85489d95-gxck4 --output=yaml | grep -A 6 resources
    resources: {}
    ...
$ kubectl get pod csi-baremetal-node-kernel-5.4-2jbtm --output=yaml | grep -A 6 resources
    resources: {}
    ...
--
    resources: {}
    ...
--
    resources: {}
    ...
--
    resources: {}
    ...
```

4. upgrade csi-operator to version with CPU/memory values:
```
export CSI_OPERATOR=0.5.0-100.019c8f4
helm upgrade csi-baremetal-operator http://asdrepo.isus.emc.com:8081/artifactory/atlantic-helm-build/csi-operator/${CSI_OPERATOR}/csi-baremetal-operator-${CSI_OPERATOR}.tgz --set image.tag=${CSI_OPERATOR} --set global.registry=asdrepo.isus.emc.com:9042 --set log.level=debug
```
5. upgrade csi driver:
```
helm upgrade csi-baremetal http://asdrepo.isus.emc.com:8081/artifactory/atlantic-helm-build/csi-operator/${CSI_OPERATOR}/csi-baremetal-deployment-${CSI_OPERATOR}.tgz --set image.tag=${CSI_VERSION} --set global.registry=asdrepo.isus.emc.com:9042 --set log.level=debug --set feature.extender=true  --set driver.drivemgr.type=halmgr  --set driver.alerts.mountConfig=true --set platform=vanilla --set  driver.controller.resources.requests.memory=50Mi --set driver.node.resources.limits.cpu=200m
```

5. check csi containers resources: 
```
anton@ubuntu:~/workspace$ kubectl get pod csi-baremetal-controller-776c84b68d-clpk6 --output=yaml | grep -A 6 resources
    resources:
      requests:
        memory: 50Mi
    ...
--
    resources: {}
    ...
--
    resources: {}
    ...
--
    resources: {}
    ...
$ kubectl get pod csi-baremetal-node-controller-6d85489d95-gxck4 --output=yaml | grep -A 6 resources
    resources: {}
    ...
$ kubectl get pod csi-baremetal-node-kernel-5.4-gmq4g --output=yaml | grep -A 6 resources
    resources: {}
    ...
--
    resources: {}
    ...
--
    resources:
      limits:
        cpu: 200m
      requests:
        cpu: 200m
    ...
--
    resources: {}

$ kubectl get pod csi-baremetal-se-tzjcf --output=yaml | grep -A 6 resources
    resources: {}
    ...
```
